unattended_scripts/install_functions/opendistro/common.sh

```diff
@@ -46,48 +46,66 @@ checkSystem() {
 
 checkNames() {
 
-    if [[ "${einame}" == "${kiname}" ]] || [[ "${einame}" == "${winame}" ]] || [[ "${kiname}" == "${winame}" ]]; then
+    if [[ -n ${einame} ]] && [[ -n ${kiname} ]] && ([[ "${einame}" == "${kiname}" ]]); then
         logger -e "The node names for Elastisearch, Kibana and Wazuh must be different."
         exit 1
     fi
 
-    if [ -n ${einame} ]; then
+    if [[ -n ${einame} ]] && [[ -n ${winame} ]] && ([[ "${einame}" == "${winame}" ]]); then
+        logger -e "The node names for Elastisearch, Kibana and Wazuh must be different."
+        exit 1
+    fi
+
+    if [[ -n ${winame} ]] && [[ -n ${kiname} ]] && ([[ "${winame}" == "${kiname}" ]]); then
+        logger -e "The node names for Elastisearch, Kibana and Wazuh must be different."
+        exit 1
+    fi
+
+    if [[ -n ${einame} ]]; then
         if [[ ! "${elasticsearch_node_names[@]}" =~ "${einame}" ]]; then
-            logger -e "The name given for the elasticsearch node does not appear on the configuration file."
+            logger -e "The name given for the ElasticSearch node does not appear on the configuration file."
             exit 1
         fi
+    fi
 
-        if [ ! -f ${base_path}/certs/${einame}.pem ] || [ ! -f ${base_path}/certs/${einame}-key.pem ]; then
-            logger -e "There is no certificate for the elasticsearch node ${einame} in ${base_path}/certs."
+    if [[ -n ${winame} ]]; then
+        if [[ ! "${wazuh_servers_node_names[@]}" =~ "${winame}" ]]; then
+            logger -e "The name given for the Wazuh server node does not appear on the configuration file."
             exit 1
         fi
-
     fi
 
-    if [ -n ${winame} ]; then
-        if [[ ! "${wazuh_servers_node_names[@]}" =~ "${winame}" ]]; then
-            logger -e "The name given for the wazuh server node does not appear on the configuration file."
+    if [[ -n ${kiname} ]]; then
+        if [[ ! "${kibana_node_names[@]}" =~ "${kiname}" ]]; then
+            logger -e "The name given for the Kibana node does not appear on the configuration file."
             exit 1
         fi
+    fi
 
-        if [ ! -f ${base_path}/certs/${winame}.pem ] || [ ! -f ${base_path}/certs/${winame}-key.pem ]; then
-            logger -e "There is no certificate for the wazuh server node ${winame} in ${base_path}/certs."
+}
+
+checkPreviousCertificates() {
```

Comment:
Is this method really necessary? If the certs folder exists, the script will report it with a:

```
if [ -d ${base_path}/certs ]; then
        logger -e "Folder ${base_path}/certs exists. Please remove the certificates folder if you want to create new certificates."
        exit 1
else
        mkdir ${base_path}/certs
fi
```

Reply:
Yes, the routines of this function were incorporated in a previous PR within the checkNames function. Only after correcting checkNames. Separate checkNames from checkPreviousCertificates.

Reply:
But then what happens if elasticsearch and kibana are installed in the same node, but in different runs?

- Run script to install elasticsearch
- Run script to install kibana

The elasticsearch installation will be successful as there will be no certificates (cert folder), but when the script to install kibana is run, it will detect that the kibana certificate exists (and if it doesn't) it will detect that the certs folder exists. Which will prevent installing a component if there is already another installed.

Reply:
These routines come from a previous merge. https://github.com/wazuh/wazuh-packages/commit/0fed0e20bb10d44677dfefbfb036e83e6cc03118

I understand that this is the expected performance. Keep the certificates generated in a previous run. In order not to lose an installation. I understand a general improvement in the management and handling of certificates is being worked on.

I invoke @alberpilot and @DFolchA to confirm.

Reply:
The unattended installer only creates certificates in the following cases:
a) If the user runs the installer with `-c` flag. This install nothing. Take the `config.yml` and make all certificates in a `certs` folder. 
b) If the user runs the installer with AIO. 

The mentioned code snipped belongs to the function `createCertificates`, which is only executed in AIO or `-c` cases.